### DISCUSSION TITLE
Write a rust-bitcoin cookbook

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#2089] [General]
I'd love to have something like a cookbook:

- This is how you receive data over P2P
- This is how you parse blocks and transactions
- This is how you construct and sign transactions
    ...

This should be well-visible at front page (README, doc) and link to separated doc. Otherwise the front page could become cluttered with things.

Originally [suggested by](https://github.com/rust-bitcoin/rust-bitcoin/issues/1463#issuecomment-1349699022) @Kixunil:

**Comment by Kixunil** [comment]
Might be nice to have a website with all the bells and whistles at some point too. Something like serde.rs :)

**Comment by VeronicaTee** [comment]
I'd like to take this up. Are there specifically preferred topics that should be included with the above-listed 3?

**Comment by Kixunil** [comment]
I think signing is the most useful of them because it's not trivial, so showing it off can be very helpful.

**Comment by VeronicaTee** [comment]
Noted. Thank you.

**Comment by apoelstra** [comment]
cc #1698 which is relevant to signing

**Comment by nph4rd** [comment]
I'd like to contribute to this too! @VeronicaTee - how're you doing with it? May I be of any help?

**Comment by VeronicaTee** [comment]
Great @arturomf94 I haven't had the chance to do much on this. I'm sure we can work something out.

**Comment by realeinherjar** [comment]
I am also interested in helping

**Comment by tcharding** [comment]
Sweet, this definitely needs some attention. The two signing tasks from the recent TABConf rust-bitcoin workshop can go in. Here is a link to the website repo where the cookbook is hosted: https://github.com/rust-bitcoin/www.rust-bitcoin.org

**Comment by realeinherjar** [comment]
@tcharding what do you think about moving this to "Discussions"? I don't think it is an "issue".

I will start working on that this weekend.
I have some non-nym experience in technical user tutorials and docs (and a little in `mdbook`) that could be useful.

**Comment by tcharding** [comment]
Moved to discussion as suggested, ee can/should IMO include stuff from all crates in our stack bech32/secp/hashes/bitcoin/miniscript.

**Comment by realeinherjar** [comment]
Work has been started in https://github.com/rust-bitcoin/www.rust-bitcoin.org/pull/9
I've managed to have automated tests for all code in the `.md` `mdbook` files.
I'll add TABConf content with good explanations soon.

**Reply by sanket1729** [reply]
Awesome. Looking forward to this.
